packages/db/src/query/optimizer.ts

```diff
@@ -424,14 +432,44 @@ function isRedundantSubquery(query: QueryIR): boolean {
  * ```
  */
 function splitAndClauses(
+  whereClauses: Array<Where>
+): Array<BasicExpression<boolean>> {
+  const result: Array<BasicExpression<boolean>> = []
+
+  for (const whereClause of whereClauses) {
+    // Skip residual WHERE clauses to prevent them from being pushed down again
+    if (isResidualWhere(whereClause)) {
+      const clause = getWhereExpression(whereClause)
+      result.push(clause)
+      continue
+    }
+
+    const clause = getWhereExpression(whereClause)
+    if (clause.type === `func` && clause.name === `and`) {
+      // Recursively split nested AND clauses to handle complex expressions
+      const splitArgs = splitAndClausesBasic(
+        clause.args as Array<BasicExpression<boolean>>
+      )
+      result.push(...splitArgs)
+    } else {
+      // Preserve non-AND clauses as-is (including OR clauses)
+      result.push(clause)
+    }
+  }
+
+  return result
+}
+
+// Helper function for recursive splitting of BasicExpression arrays
+function splitAndClausesBasic(
```

Comment:
Why do we have 2 versions (`splitAndClauses` and `splitAndClausesBasic`) of this function?
Inside `splitAndClauses` we may call into `splitAndClausesBasic` which will recursively split AND expressions but won't ignore residual where clauses. Is this safe? Can't we have the case where the `splitAndClausesBasic` will bump on a residual where clause?